File: add_first_n_odd_int.js
#!/usr/bin/env node
// add first n odd integers

function addOddIntegers(n,counter,next,accumulator) {
  if (counter === n) {
    return accumulator;
  } else {
    console.log(next);
    return addOddIntegers(n,counter+1,next+2,accumulator+next);
  }
}

function addOddIntWrapper(n) {
  return addOddIntegers(n,0,1,0);
}

console.log(addOddIntWrapper(9));

File: add_first_n_odd_int.py
#!/usr/bin/env python
# add first n odd integers

def add_odd_integers(n):
  def _add_odd_int(n,counter,next,acc):
    if counter == n:
      return acc
    else:
      return _add_odd_int(n,counter+1,next+2,acc+next)
  return _add_odd_int(n,0,1,0)

print add_odd_integers(9)